### PR TITLE
Separate Rust HTTP init from megazord init

### LINKED ISSUE #967: Gradle warning when building megazords: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'
We use a `compile` dependency on viaduct in the megazords in addition to the `api` dependency. It's required to make things compile (neither `api` nor `implementation` work), I don't know how to get rid of it nor do I fully understand it.

In order to fix the issue in request, we landed it as is, since manual testing shows it not to be a problem. We should fix it though

CC @ncalexan who might have insight.

Example of the badness: https://github.com/mozilla/application-services/blob/278dcf5cd53c4e47c1d1f32af38c3cd757db5b36/megazords/fenix/android/build.gradle#L87-L88

### LINKED ISSUE #1066: Separate HTTP init from megazord init
From slack:

> whawkins another option is for us to move the http init into its own thing (it initially was going to be, but grisha and some others thought that the current way is a nicer API, which I agree with). this would make megazord init very cheap. you'd then need to init the logger and networking at some point before they're actually needed
> but that's probably not *too* bad, although it could be a little tricky

This is essentially to reduce (well, delay) the cost of initializing the megazord / JNA.

Edit: Sorry unrelated people I just notified!

┆Issue is synchronized with this [Jira Story](https://jira.allizom.org/browse/SYNC-63)
┆Story Points: 1
┆Sprint: SYNC - 2019-07-19

### PR DESCRIPTION
**Pull request by thomcc**
Fixes #1066
Fixes #967 too, although that's not the point.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `cargo test --all` produces no test failures
  - `cargo clippy --all --all-targets --all-features` runs without emitting any warnings
  - `cargo fmt` does not produce any changes to the code
  - `./gradlew ktlint detekt` runs without emitting any warnings
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
    - I'll check this box when I finish manual testing.
- [x] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by thomcc** [comment]
I've written https://gist.github.com/thomcc/1be38c778d6ea408eb64232f0f2dac39 as a way for @hawkinsw to test to see if this can fix the performance issues we're seeing.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,47 @@
...
+            implementation "org.mozilla.appservices:httpconfig:${app_services_version}"
+            // ...
+        }
+        ```
+
+        In Kotlin:
+        ```kotlin
+        import mozilla.appservices.httpconfig.RustHttpConfig
+        // At some point after megazord init.
+        RustHttpConfig.setClient(lazy { /* your http client here */ })
```
Comment (vladikoff) [inline CHANGELOG.md]: silly question, what happens if the consumers forget to do this?

Comment (hawkinsw) [reply CHANGELOG.md]: I do not think that this is a silly question -- I think that this is exactly what we are going to have to guard against when using this in Fenix.

Comment (thomcc) [reply CHANGELOG.md]: All network requests fail with a network error, but nothing worse than that. Stuff should start working once you get around to calling RustHttpConfig.setClient.

**Comment by bug-mirror** [comment]
@hawkinsw waiting for you to confirm if this helps

**Comment by thomcc** [comment]
Subsumed by #1103

**Pull request closed without merging**